### ISSUE TITLE
Data preparation: from 1 .su file to 4 .sgy/.segy files 

### ISSUE THREAD

**Issue by calebee310** [newbee-ML/MIFN-Velocity-Picking#1]
Dear Mr. Wang,

I'm Sungyong Lee, a master student in Nagoya University, Japan.
I would like to ask about the details of the data preparation.
Since it's my first time to test the DL model with seismic data, please be patient with my silly questions.

I'm using a 1-D synthetic seismic reflection data for testing the MIFN-Velocity-Picking model. I have .su file (it's gather image and spectrum image are attached) as input seismic data, and .dat file of depth-velocity as label data.
According to the instruction, it requires segy, vel.stk.sgy, vel.pwr.sgy, vel.gth.sgy,and t_v_labels.dat files. I know .su can be converted to .sgy file, but I'm wondering why there are 3 .sgy files with 1 .segy file. 

Please check my attached the gather and spectrum images for .su file, and suggest me how can I convert it to those 4 files, as required by the instruction. And if you're willing, please also explain about those 4 files are; maybe with example images, for my understanding of the data used in the model.

I'm looking forward your reply.

Sincerely,

Sungyong Lee
![layer0001_waveform](https://user-images.githubusercontent.com/96369212/222026682-06dfb19d-0e5b-4235-afc1-c645b5116e58.JPG)
![layer0001_image](https://user-images.githubusercontent.com/96369212/222026690-8bb10142-2a99-4974-bed1-e114b0831bbf.JPG)
[z-v_label0001_dat_in_txt.txt](https://github.com/newbee-ML/MIFN-Velocity-Picking/files/10856526/z-v_label0001_dat_in_txt.txt)

**Comment by newbee-ML** [comment]
To Mr. Lee:

I'm so glad that you are interested in our work!
In this work, it requests three parts of data: 1) velocity spectrum 2) stacked gather slices 3) manually labeled stack velocity points.
Concretely, velocity spectrum and manually labeled stack velocity points are as you know. But stacked gather slices are obtained in the next seismic data processing step, where a few constant velocities are used to stack the gather using normal moveout correction. More details are showcased in Paper:   https://ieeexplore.ieee.org/document/9815261.

**Comment by calebee310** [comment]
Dear Mr.Wang,

I really appreciate your quick reply!
I see, we need 1) velocity spectrum, 2) stacked gather slices, and 3) manually labeled stack velocity points. 
I think 3) is derived from "t_v_labels.dat", but what are the others ("segy", "vel.stk.sgy", "vel.pwr.sgy", "vel.gth.sgy") related with 1) and 2)?

Maybe I couldn't understand your answers correctly, because I don't know about .sgy file. I'm sorry that I have no experience of dealing with .sgy before; I only used .su file in Seismic Unix to do seismic data processing. 
Normally, with Seismic Unix, there is only one .su file needed for visualize 1) velocity spectrum. I'm not sure how it work with .sgy file. Thus, in my understanding, it should be one .su file (or .sgy file) for each label of .dat file. Please correct my misunderstanding.

By the way, here is more information about example .su file I attached above. 

40 traces:                    # Number of traces
tracl    1 40 (1 - 40)        # Trace number
trid     1                    # Data type
offset   50 2000 (50 - 2000)  # Offset distance [m]
ns       1024                 # Number of samples in the trace
dt       4000                 # Sampling time [µs (microsecond)]

**Comment by calebee310** [comment]
P.S.

I'm sorry to disturb your time by asking the basic knowledge.
In my opinion, it would be helpful for you to provide an example file for each "segy", "vel.stk.sgy", "vel.pwr.sgy", "vel.gth.sgy", and "t_v_labels.dat" (through email or Google Drive sharing). There is no better way to understand about the format of those files than actually looking it. I'm afraid it would be hard to provide the credential data, but I think it would take more than several comments, only dealing with the data preparation part of the model...

Just in case, here is my email.
sungyonglee@seis.nagoya-u.ac.jp
or 
sungyong.caleb.lee@gmail.com

**Comment by newbee-ML** [comment]
I'm sorry, but our data is a state secret, so that we cannot share with you.
There is another solution, that you can understand the model totally, and change the pytorch dataloader defined in our codes.

**Comment by calebee310** [comment]
Dear Mr.Wang,

Thank you for sharing your time to answer my question.
I understand that the data is confidential and thus cannot be shared.
Ok, I'll try my best to understand the code and the model, though I'm not confident whether I can totally understand it or not.

Please tell me what do "vel.stk", "vel.pwr", and "vel.gth" symbolize. Maybe vel.stk = velocity stacking, and vel.gth = velocity gather. I have no idea about vel.pwr.

Please forgive me to delete my last comment as it contains my personal information.

**Comment by newbee-ML** [comment]
vel.stk = stack gather slices
vel.gth = common-midpoint (CMP) gather
vel.pwr = velocity spectrum

more information about the above concepts can be searched in wiki.

**Comment by calebee310** [comment]
Dear Mr.Wang,

Thank you so much for telling the details!
Your answers helped me a lot.
